lib/netext/httpext/transport.go

```diff
@@ -164,8 +168,33 @@ func (t *transport) measureAndEmitMetrics(unfReq *unfinishedRequest) *finishedRe
 			tags["ip"] = ip
 		}
 	}
+	var failed float64
```

Comment:
I read your comments in the PR description about the cloud output, that you can derive the state of the HTTP request by the `expected_response` tag. And while that's true, I think we should limit the magic and also send `metrics.HTTPReqFailed.Name: 0/1` in the map here, for exactly the same reason we send `http_reqs`: https://github.com/loadimpact/k6/blob/09ea784d59611d46b071b205cb2ba2d1521a8199/stats/cloud/data.go#L115-L130

After all, users would likely be setting thresholds on this metrics, we should make it possible to do so in the cloud as well.

Reply:
AFAIK the cloud has very custom logic on parsing those, which means that just adding it here *will* not help. There will still need to be changes to the code that handles it. 

Arguably it is probably better to add it, but I decided against it for two reasons:
1. it adds nothing and now there are counter, trend and Rate metrics here with absolutely no distinction between them that isn't "magical" - the cloud just knows.
2. unlike all the other metrics HTTPReqFailed can be disabled. Even if we revert this functionality this is something that we intend on readding with #1321 ... From that perspective maybe having it here when it is enabled and emitted is better as it will be future proof for when it probably shouldn't be here, and the cloud should not make it out of the tag :shrug: 

I am not hard against it by far ... it just seems like something that we will just add so it's added ... not like something that will be needed

Reply:
It's about consistency... And, for that matter, the `expected_response` tag can be disabled, so `http_req_failed` needs to be in the `http_req_li_all` map.